test/Bounty.test.js

```diff
@@ -1,113 +1,89 @@
-          let reClaimedBounty = await bounty.claimed.call();
-          assert.isFalse(reClaimedBounty);
-        }
-        try {
-          await bounty.withdrawPayments({ from: researcher });
-          assert.isTrue(false); // should never reach here
-        } catch (err) {
-          const updatedBalance = await ethGetBalance(bounty.address);
-          assert.equal(reward, updatedBalance.toNumber());
-        }
-      };
-      await bounty.createTarget({ from: researcher });
-      await awaitEvent(event, watcher);
+      const result = await this.bounty.createTarget({ from: researcher });
+      const event = await expectEvent.inLogs(
+        result.logs,
+        'TargetCreated'
+      );
+
+      const targetAddress = event.args.createdAddress;
+      await sendReward(owner, this.bounty.address, reward);
+
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
+
+      await assertRevert(
+        this.bounty.claim(targetAddress, { from: researcher }),
+      );
     });
   });
 
   describe('Against broken contract', function () {
+    beforeEach(async function () {
+      this.bounty = await InsecureTargetBounty.new();
+    });
+
     it('claims reward', async function () {
-      let owner = accounts[0];
-      let researcher = accounts[1];
-      let reward = web3.toWei(1, 'ether');
-      let bounty = await InsecureTargetBounty.new();
-      let event = bounty.TargetCreated({});
-
-      let watcher = async function (err, result) {
-        event.stopWatching();
-        if (err) { throw err; }
-        let targetAddress = result.args.createdAddress;
-        await sendReward(owner, bounty.address, reward);
-
-        const balance = await ethGetBalance(bounty.address);
-        assert.equal(reward, balance.toNumber());
-
-        await bounty.claim(targetAddress, { from: researcher });
-        let claim = await bounty.claimed.call();
-
-        assert.isTrue(claim);
-
-        await bounty.withdrawPayments({ from: researcher });
-        const updatedBalance = await ethGetBalance(bounty.address);
-        assert.equal(0, updatedBalance.toNumber());
-      };
-      await bounty.createTarget({ from: researcher });
-      await awaitEvent(event, watcher);
+      const result = await this.bounty.createTarget({ from: researcher });
+      const event = await expectEvent.inLogs(result.logs, 'TargetCreated');
+
+      const targetAddress = event.args.createdAddress;
+      await sendReward(owner, this.bounty.address, reward);
+
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
+
+      await this.bounty.claim(targetAddress, { from: researcher });
+      const claim = await this.bounty.claimed.call();
+
+      claim.should.eq(true);
+
+      await this.bounty.withdrawPayments({ from: researcher });
+      const updatedBalance = await ethGetBalance(this.bounty.address);
+      updatedBalance.should.be.bignumber.eq(0);
```

Comment:
We should also test:

a) that the researcher got the reward (curr - prev balances = reward)
b) that the bounty is no longer payable after claiming